### DISCUSSION TITLE
dropping speedify connections in 0.99.9 on slate

### DISCUSSION THREAD

**Discussion by meassick** [SmoothWAN/SmoothWAN#115] [General & Help]
I have an ethernet connection and a wireless connection being combined in Speedify.  This seems to work just fine in version 0.99.8  which seems very stable.  However in version 0.99.9 invariably one of the two connections will disconnect and be greyed out in the status window after a period of time.  I have re-installed the firmware from scratch and it makes no difference.  If I use only one or the other connection, it stays connected to that single connection.

**Comment by TalalMash** [comment]
Was the metric set for the wireless connection?

**Reply by meassick** [reply]
I tried both the default and 10 for the gateway metric in interfaces>wwan with the same results.   On version 9.99.8 it seems to run fine with it at 0 or 10.

**Reply by TalalMash** [reply]
Huh interesting, try a metric of 120.
After that, got to services -> terminal, login with root username as your usual webui password.
Then post the results of `ip route`

**Reply by meassick** [reply]
So I reinstalled 0.99.9 with a metric of 120 for wwan via UBOOT to make sure it was a clean install.  For a while both connections were OK.  Upon a reboot, only the wwan connection was there, the dsl ethernet connection did not show up in speedify at all.  In network>interfaces, PORTONE has the error message: "Error: Unknown error (DEVICE_CLAIM_FAILED)"

root@SmoothWAN:~# ip route
0.0.0.0/1 dev connectify0 scope link 
default via 192.168.10.1 dev wlan0 proto static src 192.168.10.169 metric 120 
10.202.0.0/24 dev connectify0 proto kernel scope link src 10.202.0.2 
123.123.123.123 via 192.168.10.1 dev wlan0 
128.0.0.0/1 dev connectify0 scope link 
172.17.17.0/24 dev br-lan proto kernel scope link src 172.17.17.2 
192.168.10.0/24 dev wlan0 proto static scope link metric 120 
root@SmoothWAN:~# 

rebooting a number of times, the ethernet connection is there sometimes with:
root@SmoothWAN:~# ip route
0.0.0.0/1 dev connectify0 scope link 
default via 192.168.2.1 dev PortOne proto static src 192.168.2.189 metric 10 
default via 192.168.10.1 dev wlan0 proto static src 192.168.10.169 metric 120 
10.202.0.0/24 dev connectify0 proto kernel scope link src 10.202.0.2 
123.123.123.123 via 192.168.10.1 dev wlan0 
128.0.0.0/1 dev connectify0 scope link 
172.17.17.0/24 dev br-lan proto kernel scope link src 172.17.17.2 
192.168.2.0/24 dev PortOne proto static scope link metric 10 
192.168.10.0/24 dev wlan0 proto static scope link metric 120 
root@SmoothWAN:~# 

I also tried a metic of 120 for PORTONE and also for LAN with no difference.  Also, sometimes the wireless connection does not connect instead of the ethernet connection.

Below are system and kernel logs
[syslog.txt](https://github.com/SmoothWAN/SmoothWAN/files/10832257/syslog.txt)
[kernellog.txt](https://github.com/SmoothWAN/SmoothWAN/files/10832260/kernellog.txt)

**Reply by TalalMash** [reply]
Yeah something is wrong with the init scripts from Engarde or TinyFEC when Speedify init script disables them.
As a workaround, go to System -> Startups -> Local Startup
Add before exit 0 the following line:
`/etc/init.d/network restart &`
E.g: 
![image](https://user-images.githubusercontent.com/96490382/221378878-22238acf-cb2e-41ad-a572-5aa051f85ad9.png)
Thanks for the report, added to issues!

**Reply by meassick** [reply]
still no luck, now it only stays connected to the wifi source for a couple of seconds.

**Reply by TalalMash** [reply]
Do the following in terminal just to make sure:
```
mv /etc/init.d/engardeconf /root
mv /etc/init.d/tinyvpnconf /root
```
The reboot

**Reply by meassick** [reply]
still does not work.   Did a fresh install, only speedify enabled, did modifications you recommended, tried both 10 and 120 metrics 
On PORTONE get this: "Error: Unknown error (DEVICE_CLAIM_FAILED)" and the ethernet connection is not there.  Rebooting can cause it to go away after a number of reboots and then it works.

**Comment by TalalMash** [comment]
Issues seems to be similar to this: https://github.com/openwrt/openwrt/issues/7795#issuecomment-1296469837 when using Wi-Fi clients

**Reply by TalalMash** [reply]
Are you using auto channel?
Post your `/etc/config/wireless` and `/etc/config/network` with the credentials removed.

**Reply by meassick** [reply]
yes

**Comment by TalalMash** [comment]
Got some time to test, try this:
```
cd /etc/hotplug.d/net/
rm macs
mv 00-rename 30-rename
```
Restart.

**Reply by meassick** [reply]
no change

**Comment by meassick** [comment]
yes

**Reply by meassick** [reply]
yes

**Comment by meassick** [comment]
No change

**Comment by meassick** [comment]
Also, it seems that the terminal interface automatically logs out after a number of seconds.  This was not the case in 0.99.8

**Comment by meassick** [comment]
I tried various off/on delays in the startup file which do not make a  difference:

sleep 20 &
wifi down radio0 &
wifi down radio1 &
sleep 10 &
wifi up radio0 &
sleep 2 &
wifi up radio1 &
sleep 10 &
ubus call network.interface.PortOne down &
ubus call network.interface.PortTwo down &
sleep 10 &
ubus call network.interface.PortOne up &
sleep 2 &
ubus call network.interface.PortTwo up &

**Comment by TalalMash** [comment]
Not sure of how to replicate your issue without the configuration files.

**Reply by meassick** [reply]
attached are the system and kernel logs and also the configuration that I exported.
[kernellog.txt](https://github.com/SmoothWAN/SmoothWAN/files/10834567/kernellog.txt)
[syslog.txt](https://github.com/SmoothWAN/SmoothWAN/files/10834568/syslog.txt)
[backup-SmoothWAN-2023-02-26.tar.gz](https://github.com/SmoothWAN/SmoothWAN/files/10834569/backup-SmoothWAN-2023-02-26.tar.gz)

**Comment by TalalMash** [comment]
Released a new hotfix, check it out.
I was able to replicate your issue with the backup, it had to do with the delay when using auto as the radio was scanning for DFS channels before using certain channels, this caused issues with the NSS processor when using hotplug for renaming interfaces.

**Reply by meassick** [reply]
So far so good.  I have tried two ethernet connections and one ethernet/wifi connection and both seem to work.  Also have installed Tailscale and it seems to work except for the services>terminal which gives the error "IP refused to connect" where IP is the Tailscal IP address of the Slate

**Reply by TalalMash** [reply]
Yes, an alternative is to SSH when over Tailscale or even direct, since ttyd isn't stable over websocket in recent browsers (not sure about FF)
The terminal app is for last resort. (e.g not using your daily driver computer)
Added a potential solution in the zerotier thread

**Reply by meassick** [reply]
One oddity I noticed is that the access to the status page via Tailscale works fine if I use an android device.  However, using a windows 11 PC via Tailscale with multiple different browsers (Chrome, Edge), the status page in the Speedify UI window shows "restart speedify service", even in fullscreen mode.  If I log in at the same time with the android device it shows the proper speedify information.

The same is true for the terminal window - it works fine with access via android, but not with a Windows PC.

**Reply by TalalMash** [reply]
> One oddity I noticed is that the access to the status page via Tailscale works fine if I use an android device. However, using a windows 11 PC via Tailscale with multiple different browsers (Chrome, Edge), the status page in the Speedify UI window shows "restart speedify service", even in fullscreen mode. If I log in at the same time with the android device it shows the proper speedify information.
> 
> The same is true for the terminal window - it works fine with access via android, but not with a Windows PC.

Check your firewall (e.g network adapter set to private etc) although it shouldn't matter by default, does that happen with a different Windows PC? I couldn't replicate it

**Reply by meassick** [reply]
These problem on the PC is isolated to that one, I will have to investigate it more

**Reply by meassick** [reply]
Hot Fix 3 seems to be working.  I have tried both Tailscale and Zerotier and both seem to function correctly.  I will keep an eye out to see if there are memory leaks.

**Reply by TalalMash** [reply]
HTTPS disabled in latest release (0.99.9HF4) which fixes most issues includding TTYD

**Reply by TalalMash** [reply]
Im currently facing an issue with Github, reuploading the file, seems to be GH service down.

**Reply by TalalMash** [reply]
Uploaded files seem to be visible now.

**Reply by TalalMash** [reply]
Ok so the visible file is the outdated one, I'm not sure of what's happening with Github, re-uploading again, these are the proper checksums:
```
03df098c13396d61169d472837c230a57791608f1712383b4fc42c979ded7b59 *config.buildinfo
44c146f403dc247eafcc7d7dfb1f934f8b59061d8fb911c8947cda52ec4d7eba *feeds.buildinfo
a862c4d6c899b64c1f39f52b432cff293667c0289070e3e89a1f0896f43fec0b *openwrt-ipq807x-glinet_ax1800-initramfs-kernel.bin
59f6a275466f5776070cddb1c45ce892e41e8e79ae177c9f710d4ce80a890d1d *openwrt-ipq807x-glinet_ax1800-squashfs-nand-factory.img
928dcd2fb8b9fa55b8875ed7e3ab3504d112ba2c830e799c595e6f75946fb9e1 *openwrt-ipq807x-glinet_ax1800-squashfs-sysupgrade.tar
a551a9617f363321f591e7d6236a7b8818d6f41fd972d87c128f727deed3992f *openwrt-ipq807x-glinet_ax1800.manifest
6d9d01f8dd06aed73f3d3cbfc70cb31a5213e641fcdcbf9c947bdf6480d8b2c5 *openwrt-ipq807x-glinet_axt1800-initramfs-kernel.bin
92e636d7203b5345fcf0e9556bb75e0d5aff29622695c7e0c7ff76d806619a7d *openwrt-ipq807x-glinet_axt1800-squashfs-nand-factory.img
8bb06b429c9c4ab68a9651b2531b17e152f94cde498a197c7f78aa060846b81d *openwrt-ipq807x-glinet_axt1800-squashfs-sysupgrade.tar
3ee425d91936a6c7c70c34b6c544cbcecdb51fe34f8bc932f836cec85c9a872d *openwrt-ipq807x-glinet_axt1800.manifest
53806acb13883123030d11b1808456e1cf8748d84d83c4161006eaecd7c2f5fb *version.buildinfo
```

**Comment by meassick** [comment]
Yes, both the Slate and Flint versions are there now
